File: FastMines_core/src/fmg/common/ui/UiInvoker.cs
using System;
using Fmg.Core.Img;

namespace Fmg.Common.UI {

    /// <summary> Factory of UI timers/animators/deferred invokers </summary>
    public static class UiInvoker {

        /// <summary> Deferred execution in the UI thread </summary>
        public static Action<Action> Deferred = doRun => {
            throw new NotImplementedException();
        };

        /// <summary> Platform-dependent factory of <see cref="IAnimator"/>. Set from outside... </summary>
        public static Func<IAnimator> Animator = () => {
            throw new NotImplementedException();
        };

        /// <summary> Platform-dependent factory of <see cref="ITimer"/>. Set from outside... </summary>
        public static Func<ITimer> TimerCreator = () => {
            throw new NotImplementedException();
        };

    }

}

File: FastMines_core/src/fmg/common/ui/UiInvoker.java
package fmg.common.ui;

import java.util.function.Consumer;
import java.util.function.Supplier;

import fmg.core.img.IAnimator;

/** Factory of UI timers/animators/deferred invokers */
public final class UiInvoker {
    private UiInvoker() {}

    /** Deferred execution in the UI thread */
    public static Consumer<Runnable> Deferred = run -> {
        throw new UnsupportedOperationException("Not implemented...");
    };

    /** Platform-dependent factory of {@link IAnimator}. Set from outside... */
    public static Supplier<IAnimator> Animator = () -> {
        throw new UnsupportedOperationException("Not implemented...");
    };

    /** Platform-dependent factory of {@link ITimer}. Set from outside... */
    public static Supplier<ITimer> TimeCreator = () -> {
        throw new UnsupportedOperationException("Not implemented...");
    };

}
